src/emqx_client.erl

```diff
@@ -135,23 +154,47 @@
 
 -type(subscribe_ret() :: {ok, properties(), [reason_code()]} | {error, term()}).
 
--export_type([client/0, topic/0, qos/0, properties/0, payload/0,
-              packet_id/0, pubopt/0, subopt/0, reason_code/0]).
+-type(request_input() :: binary()).
 
-%% Default timeout
--define(DEFAULT_KEEPALIVE,       60000).
--define(DEFAULT_ACK_TIMEOUT,     30000).
--define(DEFAULT_CONNECT_TIMEOUT, 60000).
+-type(response_payload() :: binary()).
 
--define(PROPERTY(Name, Val), #state{properties = #{Name := Val}}).
+-type(request_handler() :: fun((request_input()) -> response_payload())).
+
+-type(response_topic() :: ?SHARED(ShareGroup :: binary(), binary())).
+
+-export_type([client/0, topic/0, qos/0, properties/0, payload/0,
```

Comment:
it's a good practice to define `-export_type` right after the function `-export` section.